src/components/UserAvatar.vue

```diff
@@ -1,15 +1,15 @@
-<script setup>
+<script setup lang="ts">
 import { watchEffect } from 'vue';
 import { shorten, explorerUrl } from '@/helpers/utils';
 import { useUsername } from '@/composables/useUsername';
 
-const props = defineProps({
-  address: String,
-  space: Object,
-  proposal: Object,
-  profile: Object,
-  onlyUsername: Boolean
-});
+const props = defineProps<{
+  address: string;
+  space: Record<string, any>;
```

Comment:
Maybe we should also think about typing the space and proposal property instead of using Record. 

Something to keep in mind for later.

Reply:
Added some types :) Please review again @midgerate